src/compiler/crystal/semantic/restrictions.cr

```diff
@@ -358,16 +358,32 @@ module Crystal
       stricter_pair_to_num(self_stricter, other_stricter)
     end
 
+    def old_min_args_size
+      if splat_index = self.def.splat_index
+        args = self.def.args
+        unless args[splat_index].name.empty?
+          default_value_index = args.index(&.default_value)
+          min_size = default_value_index || args.size
+          min_size -= 1 unless default_value_index.try(&.< splat_index)
+          return min_size
+        end
+      end
+      self.min_size
+    end
```

Comment:
This implementation looks very similar to `Def#min_max_args_sizes`. Could we just use that?

Reply:
First the max size is unaffected, second we already know that `-Dpreview_overload_order` is not in effect at this point. If we check for that flag in `min_max_args_sizes` itself there could be a performance impact _and_ some other uses of the method also become inconsistent (e.g. too-few-argunents errors).

There might be a better way to derive the old min size from the new min size. I don't know if it would end up doing the same amount of work though.